packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -364,23 +364,22 @@ export default function(
         name,
         name,
       );
-    }
-
-    const state = instance.state;
-    if (state && (typeof state !== 'object' || isArray(state))) {
-      warning(
-        false,
-        '%s.state: must be set to an object or null',
-        getComponentName(workInProgress),
-      );
-    }
-    if (typeof instance.getChildContext === 'function') {
-      warning(
-        typeof type.childContextTypes === 'object',
-        '%s.getChildContext(): childContextTypes must be defined in order to ' +
-          'use getChildContext().',
-        getComponentName(workInProgress),
-      );
+      const state = instance.state;
+      if (state && (typeof state !== 'object' || isArray(state))) {
```

Comment:
This is indentation change to make DEV-only intent clear (even though the whole function is only called in DEV).
But I also made it use `name`.

Reply:
https://github.com/facebook/react/pull/12429/files?w=1 without whitespace changes